### PR TITLE
Issue #3850: Fixed "'typecast' is not followed by whitespace" for multiline cast.

### PR DESCRIPTION
**Pull request by soon**
Issue #3850 

Diff: https://soon-checkstyle-diffs.github.io/

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java [hunk 1]
```diff
@@ -85,6 +85,8 @@ private int nonVoid()
     private void testCasts()
     {
         Object o = (Object) new Object(); // ok
+        o = (java.lang.
+                Long) o; // ok
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: Please add an error case that spans lines to show it still reports violation correctly.

Comment (soon) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: @rnveach Added error case

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java [hunk 2]
```diff
@@ -85,6 +85,10 @@ private int nonVoid()
     private void testCasts()
     {
         Object o = (Object) new Object(); // ok
+        o = (java.lang.
+                Long)o; // error
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: "error" ==> "violation".
severity of event is not under attention here.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java [hunk 3]
```diff
@@ -85,6 +85,10 @@ private int nonVoid()
...
+        o = (java.lang.
+                Long) o; // ok
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: I do not like amount of tests are updated , so you force me to verify all changes, too much.
Please create separate input file with with cases, or move them to the end of this file to have changes only in test that are affected in functional way.

Comment (soon) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: I was thinking about moving this case into separate method or file, but then I found a method with whitespace checks for typecasts. Going to move to a separate file.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java (hunk 2 above)
Comment (soon) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java (hunk 3 above)
Comment (soon) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java]: Done

**Comment by romani** [comment]
I am fine with changes.
But we need to do regression diff testing.

Please run our groovy tool , https://github.com/checkstyle/contribution/tree/master/checkstyle-tester . All details on usage is in readme file.

Please share a report to us for verification.

**Comment by soon** [comment]
@romani Should I run the tool for guava only (as defined in `projects-to-test-on.properties`), or include other projects too? Should I share the diff somewhere or just drop here a zipped results?

For now I prepared diff report for the guava project and shared it here: https://soon-checkstyle-diffs.github.io/

Please, let me know if other checks should be made.

**Comment by romani** [comment]
> or include other projects too ?

Please include as much as possible.

>Should I share the diff somewhere or just drop here a zipped results?

There are point in readme to put it in web (github.io). You can see examples of reports in other PR.

**Comment by romani** [comment]
>For now I prepared diff report for the guava project and shared it here

Did you activate your Check in config ? I see "AbstractClassName"
https://soon-checkstyle-diffs.github.io/guava/index.html

**Comment by soon** [comment]
@romani Totally missed that, sorry. Updated link.

It found 5 differences in 3 projects (openjdk9 (1), checkstyle (2), checkstyle-with-excludes (2)). Seems like all of them are valid - all lines contains a space after typecast but the *base* version raises a warning. Could you please review the diff?

**Comment by romani** [comment]
Report is valid.

**Pull request merged**